lib/facter/resolvers/release_with_regex.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module Facter
+  module Resolvers
+    class ReleaseWithRegex < BaseResolver
```

Comment:
This is the same as `SpecificReleseFile` + a regex. Since we exposed the `options`, why not have only a resolver which does the following:
 - reads version from a options[:release_file] or a default location of options[:release_file] is not sent
 - apply a regex if options[:regex] is sent

Maybe we can also compact `ReleaseFromFirstLine` into that.

Reply:
`SpecificReleseFile` and `ReleaseWithRegex` are unified in `SpecificReleseFile`. The change is under the last commit for more visibility.